plugins/dfu/dfu-target.c

```diff
@@ -472,6 +472,42 @@ dfu_target_status_to_error_msg (DfuStatus status)
 	return NULL;
 }
 
+gboolean
+dfu_target_poll_in_manifest (DfuTarget *target, GError **error)
+{
+	DfuTargetPrivate *priv = GET_PRIVATE (target);
+	DfuStatus status;
+
+	/* get the status */
+	if (!dfu_device_refresh (priv->device, error))
+		return FALSE;
+
+	/* wait for DFU_STATE_DFU_MANIFEST to not be set */
+	while (dfu_device_get_state (priv->device)
+					== DFU_STATE_DFU_MANIFEST_SYNC ||
+			dfu_device_get_state (priv->device)
+					== DFU_STATE_DFU_MANIFEST) {
```

Comment:
so if the device is permanently stuck in this state the tool what happens to tool?  It seems it gets hung.  I don't know how common that is, but I wonder if we want an upper bound to prevent it?  It seems to not be in the spec.

Reply:
Yeah, you are right. If something is wrong with the reprogramming, the device may be stuck in this state. We can limit the max tries here, but not sure what value should be assigned. It is device-specific.